src/post.js

```diff
@@ -86,7 +86,9 @@ Module["solve"] = function (model_str, highs_options) {
     "write and extract solution"
   );
   _Highs_destroy(highs);
-  const output = parseResult(stdout_lines, status);
+  const isQuadratic = model_str.indexOf("[") !== -1;
```

Comment:
Is this really correct ? Isn't there a way to detect a quadratic program from the output ?
Is there absolutely no way for a non-quadratic program to contain a `[` ?

Reply:
Yeah, good question; I pondered the same. [The specification](http://lim.univ-reunion.fr/staff/fred/Enseignement/Optim/doc/CPLEX-LP/CPLEX-LP-file-format.html) says the following:

>  Also, the following characters are not valid in variable names (in order to allow for quadratic objective information): ^, *, [ and ]. 

I believe this makes the check safe (although it'd certainly be more comfortable to be able to pick the information out of the library somehow).

Reply:
Yes, we should improve output parsing instead of trying to parse the input.

Reply:
I've added a commit changing the logic for determining the headers: Rather than basing them on the type of problem (which was what was hard for the QPs), this instead decides whether or not a particular header should be included by looking at the row immediately below, checking whether there is any data available. The data can be either left or right aligned, so we check both. I added a few more tests to cover some edge cases.

Previously, `isLinear` (and `isQuadratic`) would be used to determine which headers were there, so I can see how one might as well add the information to the output, but with this change, they're not used for much, and I do wonder if there's any benefit from having them in the output (presumably the user would typically know what problem they come with), but chances are there's a use case I'm missing?

Reply:
And thanks for the swift review!